test/deploy.py

```diff
@@ -0,0 +1,486 @@
+        self.execRemoteScript(
+            vm_name, relative_script_path,
+            ['prep_docker', self.complete_repo_path]
+        )
+
+    def bootstrapCephadm(self, image='localhost:5000/canonical/ceph:latest') -> None:
+        '''Bootstrap Cephadm using cephadm-test script.'''
+        self.execRemoteScript(
+            vm_name, 'scripts/cephadm-test.sh',
+            ['deploy_cephadm', image]
+        )
+
+    def addOSDs(self, vm_name:string, checkCount=10, expectedOSDNum=3) -> None:
+        '''Deploy OSD Daemon using cephadm-test script.'''
+        print("Adding OSDs, it may take a few minutes.")
+        status_cmd = [
+            'ceph', 'status', '-f', 'json'
+        ]
+        cmd = [
+            'ceph', 'orch', 'apply', 'osd', '--all-available-devices'
+        ]
+        self.checkOutputOnCephadmShell(vm_name, cmd)
+
+        for attempt in range(0, checkCount):
+            status = json.loads(self.checkOutputOnCephadmShell(vm_name, status_cmd))
+            osd_count = status['osdmap']['num_osds']
+            if osd_count >= expectedOSDNum:
+                break
+            print("Attempt {}: OSD not up! Count {}".format(attempt, osd_count))
+            time.sleep(60) # Wait for a minute
+
+        status = json.loads(self.checkOutputOnCephadmShell(vm_name, status_cmd))
+        osd_count = status['osdmap']['num_osds']
+        if osd_count < expectedOSDNum:
+            raise EnvironmentError("OSDs not up Count {}".format(osd_count))
+        print("OSD Count {}".format(osd_count))
+
+    def patchCephRuleForSingleNode(self, vm_name:string) -> None:
+        '''Creates a suitable replication rule for single node deployments.'''
+        # Create a new rule with OSD as failure domain.
+        new_rule_name = 'new_replication_rule_osd'
+        cmd = [
+            'ceph', 'osd', 'crush', 'rule', 'create-replicated',
+            new_rule_name, 'default', 'osd'
+        ]
+        self.checkOutputOnCephadmShell(vm_name, cmd)
+
+        # Make new rule default for all pools.
+        cmd = [
+           'ceph','osd','pool','ls', '-f', 'json'
+        ]
+        pools = json.loads(self.checkOutputOnCephadmShell(vm_name, cmd))
+        for pool in pools:
+            cmd = [
+                'ceph', 'osd', 'pool', 'set', pool, 'crush_rule',
+                new_rule_name
+            ]
+            self.checkOutputOnCephadmShell(vm_name, cmd)
+
+        # Fetch and delete other replication rules
+        cmd = [
+            'ceph', 'osd', 'crush', 'rule', 'dump', '-f', 'json'
+        ]
+        rules = json.loads(self.checkOutputOnCephadmShell(vm_name, cmd))
+        for rule in list(rules):
+            if rule['rule_name'] == new_rule_name:
+                continue
+            cmd = [
+                'ceph', 'osd', 'crush', 'rule', 'rm', rule['rule_name'],
+            ]
+            self.checkOutputOnCephadmShell(vm_name, cmd)
+
+def printScriptHelp() -> None:
+    print("Script Help")
```

Comment:
This is not a lot of help :-)

Suggest using the argparse package for automatic argument parsing and help

Reply:
:D 
This was the placeholder I forgot to implement. I had already pushed a subsequent commit with this.